src/components/ConsentBanner/ConsentBanner.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import "./ConsentBanner.css";
+
+export interface ConsentBannerProps {
+    showConsentBanner?: boolean,
+    consentText?: string,
+    children?: React.ReactNode,
+    showConsentInfoIssueLink?: boolean,
+    onAccept: () => void,
+    onDecline: () => void
+}
+
+const ConsentBanner = ({showConsentBanner, consentText, showConsentInfoIssueLink, onAccept, onDecline, children}:ConsentBannerProps ) => {
+    const msgText = consentText || "We're collecting web-vitals, pageview and other metrics in order to improve and prioritize our work on IPFS";
+    
+    if(children) return (
+        <div className="metrics-notification-wrapper">
+            {children}
+        </div>
+    )
+    
+    return showConsentBanner ? (
```

Comment:
these ternary statements in react are a big pet peeve of mine. We should use [guard clauses](https://github.com/rubocop/ruby-style-guide#no-nested-conditionals) instead. i.e.

```
if (!showConsentBanner) {
return null
}

return (<component...>)
```

in addition to being a pet peeve, ternary statements are generally a bad idea for anything and everything, except in the very few instances they're not. I can explain more in person if you like